scripts/bash/gitlab_helpers.sh

```diff
@@ -0,0 +1,393 @@
+    errMsg=$(gitlab-ctl reconfigure 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg" 
+    
+    errMsg=$(gitlab-ctl restart 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg"
+
+    logger $logName "INFO" "Updating EXTERNAL URL to CI_EXTERNAL_URL CI/CD variable"
+    errMsg=$(gitlab-rails runner "Ci::InstanceVariable.where(key: 'CI_EXTERNAL_URL').update(value: '$EXTERNAL_URL')" 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg" 
+}
+
+
+gitlab_install () {
+    local logName=$1
+    local gitlabVersion=$2
+    # Install Postfix non-interactive
+    logger $logName "INFO" "Starting postfix installation for domain: $INSTANCE_EXTERNAL_DOMAIN"
+    debconf-set-selections <<< "postfix postfix/mailname string $INSTANCE_EXTERNAL_DOMAIN"
+    debconf-set-selections <<< "postfix postfix/main_mailer_type string 'Internet Site'"
+    errMsg=$(apt-get install --assume-yes postfix 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg" 
+
+    # Install Gitlab Server
+    logger $logName "INFO" "Setting up Gitlab installation at version $gitlabVersion"
+    curl https://packages.gitlab.com/install/repositories/gitlab/gitlab-ee/script.deb.sh | bash
+    #Specific version installation:
+    logger $logName "INFO" "Downloading Gitlab from https://packages.gitlab.com/gitlab/gitlab-ee/packages/ubuntu/bionic/gitlab-ee_$gitlabVersion.0_amd64.deb/download.deb"
+    errMsg=$(wget --content-disposition https://packages.gitlab.com/gitlab/gitlab-ee/packages/ubuntu/bionic/gitlab-ee_$gitlabVersion.0_amd64.deb/download.deb 2>&1)
+    get_last_error $logName $? $ERR_ACTION_EXIT "$errMsg" 
+    logger $logName "INFO" "Installing Gitlab"
+    errMsg=$(dpkg -i gitlab-ee_$gitlabVersion.0_amd64.deb 2>&1)
+    get_last_error $logName $? $ERR_ACTION_EXIT "$errMsg"
+    
+    logger $logName "INFO" "Reconfiguring Gitlab"
+    errMsg=$(gitlab-ctl reconfigure 2>&1)
+    get_last_error $logName $? $ERR_ACTION_CONT "$errMsg"     
+}
+
+
+setup_cicd_vars () {
+    local logName=$1
+    # Set instance level environment variables, so pipelines can utilize them
+    logger $logName "INFO" "Fetching values for instance level CI/CD variables"
+    GCP_PROJECT_ID=`gcloud config list --format 'value(core.project)' 2>/dev/null`
+    INSTANCE_INTERNAL_HOSTNAME=`curl -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/hostname"`
+    INSTANCE_INTERNAL_URL=$INSTANCE_PROTOCOL://$INSTANCE_INTERNAL_HOSTNAME
+    INSTANCE_INTERNAL_API_V4_URL=$INSTANCE_PROTOCOL://$INSTANCE_INTERNAL_HOSTNAME/api/v4
+    
+    logger $logName "INFO" "CI_EXTERNAL_URL=$EXTERNAL_URL"
+    logger $logName "INFO" "CI_SERVER_HOST=$INSTANCE_INTERNAL_HOSTNAME" 
+    logger $logName "INFO" "CI_SERVER_URL=$INSTANCE_INTERNAL_URL" 
+    logger $logName "INFO" "CI_API_V4_URL=$INSTANCE_INTERNAL_API_V4_URL" 
+    logger $logName "INFO" "CONTAINER_REGISTRY_NAMESPACE=$GCP_PROJECT_ID" 
+    logger $logName "INFO" "CONTAINER_REGISTRY_HOST=gcr.io"
```

Comment:
Why not creating one message for this?

Reply:
fixed

Reply:
trying to keep it more organized, instead of providing it directly to the logger function